sklearn/preprocessing/tests/test_encoders.py

```diff
@@ -637,3 +652,314 @@ def test_categories(density, drop):
 @pytest.mark.parametrize('Encoder', [OneHotEncoder, OrdinalEncoder])
 def test_encoders_has_categorical_tags(Encoder):
     assert 'categorical' in Encoder()._get_tags()['X_types']
+
+
+@pytest.mark.parametrize("kwargs", [
+    {'max_levels': 2},
+    {'min_frequency': 11},
+    {'min_frequency': 0.29},
+    {'max_levels': 2, 'min_frequency': 6},
+    {'max_levels': 4, 'min_frequency': 12},
+])
+@pytest.mark.parametrize("categories",
+                         ["auto", [['a', 'b', 'c', 'd']]])
+def test_ohe_infrequent_two_levels(kwargs, categories):
```

Comment:
Please briefly describe this test and the others that are not obvious

Reply:
Done